### DISCUSSION TITLE
Can you add some examples of AST modifications? Convenient for us to implement some simple functions on the frontend of Rust when we are just starting out

### DISCUSSION THREAD

**Discussion by SKT1yang** [oxc-project/oxc#4653] [Q&A]
Implementation function: Modify AST, wrap the value declared by the variable with a function named "t", 
for example: `let value = "text"` is modified to `let value = t("text")`

I am currently stuck in the step of replacing nodes

<img width="1548" alt="image" src="https://github.com/user-attachments/assets/aadf8b84-89c0-409c-a5d7-2a7900933e6f">
[i18n-wrap-rust.zip](https://github.com/user-attachments/files/16490595/i18n-wrap-rust.zip)

**Comment by SKT1yang** [comment]
like this:  https://zhuanlan.zhihu.com/p/617125984?utm_medium=social&utm_psn=1803771809939734528&utm_source=wechat_session

**Comment by milesj** [comment]
You can't mutate the AST because `expr` is not mutable. Only properties on `ImportParser` are mutable.

**Reply by SKT1yang** [reply]
ok, thanks. Can you provide a minimum example of modifying a node? I searched for the source code and resources online, but couldn't find them. I am learning Rust and trying to use OXC to implement some small tools and functions.

**Reply by milesj** [reply]
If you're writing your own code and this isn't an oxc linter/transformer/etc, you should use the `VisitMut` trait instead of `Visit`. This allows the node to be mutable.

https://github.com/oxc-project/oxc/blob/main/crates/oxc_ast/src/generated/visit_mut.rs

**Reply by SKT1yang** [reply]
Okay, thank you. I'll go take a look
